lib/aws_codegen/rest_service.ex

```diff
@@ -360,21 +424,159 @@ defmodule AWS.CodeGen.RestService do
           |> Enum.filter(filter_fn(param_type))
           |> Enum.map(fn {name, x} ->
             required = Enum.member?(required_members, name)
-            build_parameter(language, {name, x["traits"][param_type]}, required)
+
+            tynfo =
+              get_type_info(x, api_spec)
+
+            docs = get_in(x, ["traits", "smithy.api#documentation"])
+
+            docs =
+              if is_nil(docs) do
+                ""
+              else
+                extract_param_docs_snippet(docs)
+              end
+
+            if is_nil(tynfo) do
+              build_parameter(language, {name, x["traits"][param_type]}, required, "string", docs)
+            else
+              build_parameter(language, {name, x["traits"][param_type]}, required, tynfo, docs)
+            end
           end)
       end
     else
       []
     end
   end
 
+  def get_body_required?(api_spec, operation_spec) do
```

Comment:
@onno-vos-dev I'm struggleing with this part. Can you chime in here? I'd like to figure out if a body is required or not.

Optional example:
```json
        "com.amazonaws.s3#CreateBucketRequest": {
            "type": "structure",
            "members": {
                [...]
                "CreateBucketConfiguration": {
                    "target": "com.amazonaws.s3#CreateBucketConfiguration",
                    "traits": {
                        "smithy.api#documentation": "<p>The configuration information for the bucket.</p>",
                        "smithy.api#httpPayload": {},
                        "smithy.api#xmlName": "CreateBucketConfiguration"
                    }
                },

```

Required example:
```json
        "com.amazonaws.s3#UploadPartRequest": {
            "type": "structure",
            "members": {
                "Body": {
                    "target": "com.amazonaws.s3#StreamingBlob",
                    "traits": {
                        "smithy.api#default": "",
                        "smithy.api#documentation": "<p>Object data.</p>",
                        "smithy.api#httpPayload": {}
                    }
                },
```

Required example:
```json
        "com.amazonaws.apigatewayv2#CreateApiRequest": {
            "type": "structure",
            "members": {
                "ApiKeySelectionExpression": {
                    "target": "com.amazonaws.apigatewayv2#SelectionExpression",
                    "traits": {
                        "smithy.api#documentation": "...",
                        "smithy.api#jsonName": "apiKeySelectionExpression"
                    }
                },
               [...]
           }
           "traits": {
                "smithy.api#documentation": "<p>Creates a new Api resource to represent an API.</p>",
                "smithy.api#input": {}
            }
```

I'm not sure if these examples are even covering all cases, and searching for `jsonName` feels very dirty, but I don't think I'm deep enough into the smithy spec to see a better way?